ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/privacy/EeaSendRawTransaction.java

```diff
@@ -89,34 +87,42 @@ public JsonRpcResponse response(final JsonRpcRequest request) {
           request.getId(), JsonRpcError.UNIMPLEMENTED_PRIVATE_TRANSACTION_TYPE);
     }
 
-    final Transaction transaction;
+    final String enclaveKey;
     try {
-      transaction = handlePrivateTransaction(privateTransaction);
-    } catch (final InvalidJsonRpcRequestException e) {
+      enclaveKey = privateTransactionHandler.sendToOrion(privateTransaction);
+    } catch (final IOException e) {
       return new JsonRpcErrorResponse(request.getId(), JsonRpcError.ENCLAVE_ERROR);
     }
 
-    final ValidationResult<TransactionInvalidReason> validationResult =
-        transactionPool.addLocalTransaction(transaction);
-    return validationResult.either(
-        () -> new JsonRpcSuccessResponse(request.getId(), transaction.hash().toString()),
-        errorReason ->
-            new JsonRpcErrorResponse(
-                request.getId(), convertTransactionInvalidReason(errorReason)));
-  }
-
-  protected long getNonce(final Address address) {
-    return blockchain.getTransactionCount(address, blockchain.headBlockNumber());
-  }
-
-  private Transaction handlePrivateTransaction(final PrivateTransaction privateTransaction)
-      throws InvalidJsonRpcRequestException {
+    final String privacyGroupId;
     try {
-      return privateTransactionHandler.handle(
-          privateTransaction, () -> getNonce(privateTransaction.getSender()));
+      privacyGroupId =
+          privateTransactionHandler.getPrivacyGroup(
+              enclaveKey, privateTransaction.getPrivateFrom());
     } catch (final IOException e) {
-      throw new InvalidJsonRpcRequestException("Unable to handle private transaction", e);
+      return new JsonRpcErrorResponse(request.getId(), JsonRpcError.ENCLAVE_ERROR);
```

Comment:
This IOException should get logged here, even it its only at TRACE level.

Reply:
this is addressed in #1455